src/main/scala/scala/collection/decorators/IteratorDecorator.scala

```diff
@@ -72,4 +72,44 @@ class IteratorDecorator[A](val `this`: Iterator[A]) extends AnyVal {
     loop(immutable.List.empty)
   }
 
+  /**
+    * Constructs an iterator where consecutive elements are accumulated as
+    * long as the output of f for each element doesn't change.
+    * <pre>
+    * Vector(1,2,2,3,3,3,2,2)
+    *     .iterator
+    *     .groupUntilChanged(identity)
+    *     .toList
+    * </pre>
+    * produces
+    * <pre>
+    * List(Seq(1),
+    *      Seq(2,2),
+    *      Seq(3,3,3),
+    *      Seq(2,2))
+    * </pre>
+    * @param f the function to compute a key for an element
+    * @tparam K the type of the computed key
+    * @return an iterator of sequences of the consecutive elements with the
+    *         same key in the original iterator
+    */
+  def groupUntilChanged[K](f: A => K): Iterator[Iterable[A]] =
```

Comment:
Sorry for the nitpick / bikeshedding but maybe something shorter like `chunkBy`?

Reply:
Other possibilities:
`segmentBy`
`splitBy`
`sliceBy`

Reply:
In fact, since this is kind of a generalization of https://github.com/scala/scala-collection-contrib/pull/6, it probably makes most sense to call it `splitBy`

Reply:
`splitBy` seems to be what [Mathematica calls it](https://reference.wolfram.com/language/ref/SplitBy.html)  sounds good to me.

Reply:
I will push a fixup commit addressing the rename shortly